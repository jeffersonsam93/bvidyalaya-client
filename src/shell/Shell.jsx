import React, { Component } from "react";
//import PropTypes from "prop-types";
//import { push } from "react-router-redux";
import { connect } from "react-redux";

import { withRouter, Route, Redirect } from "react-router-dom";
import { withStyles } from "@material-ui/core/styles";
import classNames from "classnames";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Hidden from "@material-ui/core/Hidden";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MenuIcon from "@material-ui/icons/Menu";
import shellStyle from "styles/shell/shellStyle";

import SideNav from "./parts/SideNav";

//import image from "assets/img/sidebar-2.jpg";
import logo from "assets/logo.png";

import { toggleNav } from "./state/actions";
import appRoutes from "../modules";
import HeaderLinks from "./parts/HeaderLinks";
import Setting from "./parts/Setting";

class Shell extends Component {
  activeRoute(routeName) {
    return this.props.location &&
      this.props.location.pathname &&
      this.props.location.pathname.indexOf(routeName) > -1
      ? true
      : false;
  }

  render() {
    const { classes, toggleNav, location, open, ...rest } = this.props;
    var navbarName = "/setting" === location.pathname ? "Settings" : "";
    appRoutes.map((prop, key) => {
      if (location && prop.path === location.pathname) {
        navbarName = prop.navbarName;
        return navbarName;
      }
      return null;
    });
    return (
      <div className={classes.root}>
        <SideNav
          routes={appRoutes}
          logoText={"BVPS"}
          logo={logo}
          color="blue"
          {...rest}
        />
        <AppBar
          position="absolute"
          className={classNames(classes.appBar, open && classes.appBarShift)}
        >
          <Toolbar disableGutters={!open} className={classes.toolbar}>
            <Hidden smDown>
              <IconButton
                color="inherit"
                aria-label="open drawer"
                onClick={toggleNav}
                className={classNames(classes.menuButton, open && classes.hide)}
              >
                <MenuIcon />
              </IconButton>
            </Hidden>
            <Typography variant="title" color="inherit" noWrap>
              {navbarName}
            </Typography>
            <Hidden mdDown>
              <HeaderLinks className={classes.headerLinks} />
            </Hidden>
            <Hidden mdUp>
              <IconButton
                color="inherit"
                aria-label="open drawer"
                onClick={toggleNav}
                className={classNames(classes.menuButton, open && classes.hide)}
              >
                <MenuIcon />
              </IconButton>
            </Hidden>
          </Toolbar>
        </AppBar>
        <div className={classes.content}>
          <div className={classes.toolbar} />
          <div className={classes.actualContent}>
            {appRoutes.map((route, index) => {
              if (route.redirect)
                return (
                  <Route
                    key={index}
                    path={route.path}
                    exact
                    render={() => <Redirect from={route.path} to={route.to} />}
                  />
                );
              //   return <Redirect from={route.path} exact={true} to={route.to} key={index} />;
              // Render more <Route>s with the same paths as
              // above, but different components this time.
              return (
                <Route
                  key={index}
                  path={route.path}
                  exact
                  component={route.component}
                />
              );
            })}
            <Route path="/setting" exact component={Setting} />
            {/* <Route path="*" render={() => <Redirect to="/notfound" />} /> */}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    open: state.shell.navOpen
  };
};

const mapDispatchToProps = dispatch => {
  return {
    toggleNav: () => {
      dispatch(toggleNav());
    }
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(
  withRouter(withStyles(shellStyle, { withTheme: true })(Shell))
);
