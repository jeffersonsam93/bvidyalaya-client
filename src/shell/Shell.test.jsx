import React from "react";
import ReactDOM from "react-dom";
import Shell from "./Shell";
import renderer from "react-test-renderer";
import TestWrapper from "testing";

describe("Shell", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <TestWrapper>
        <Shell />
      </TestWrapper>,
      div
    );
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <TestWrapper>
          <Shell />
        </TestWrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
