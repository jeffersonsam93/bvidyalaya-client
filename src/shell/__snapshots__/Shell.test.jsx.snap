// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shell renders correctly 1`] = `
<div
  className="Shell-root-1"
>
  <div>
    <div
      className="MuiDrawer-docked-98"
    >
      <div
        className="MuiPaper-root-45 MuiPaper-elevation0-47 MuiDrawer-paper-99 withRouter-SideNav--drawerPaper-14 withRouter-SideNav--drawerPaperClose-15 MuiDrawer-paperAnchorLeft-100 MuiDrawer-paperAnchorDockedLeft-104"
      >
        <div
          className="withRouter-SideNav--toolbar-16"
        >
          <a
            className="withRouter-SideNav--logoLink-18"
            href="/"
          >
            <div
              className="withRouter-SideNav--logoImage-19"
              style={
                Object {
                  "backgroundImage": "url(logo.svg)",
                  "backgroundRepeat": "no-repeat",
                }
              }
            />
          </a>
          <button
            className="MuiButtonBase-root-115 MuiIconButton-root-109 MuiIconButton-colorInherit-110"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiIconButton-label-114"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-117"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <g>
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </g>
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root-123"
            />
          </button>
        </div>
        <div
          className="withRouter-SideNav--sidebarWrapper-36"
        >
          <div
            className="withRouter-SideNav--list-22"
          >
            <ul
              className="MuiList-root-130 MuiList-padding-131 withRouter-SideNav--list-22"
            >
              <div
                className="MuiButtonBase-root-115 MuiListItem-root-134 MuiListItem-default-137 MuiListItem-gutters-141 MuiListItem-button-142 withRouter-SideNav--itemLink-24 withRouter-SideNav--itemLinkClose-25"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-117 MuiListItemIcon-root-144 withRouter-SideNav--itemIcon-26"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
                    />
                  </g>
                </svg>
                <div
                  className="MuiListItemText-root-145 withRouter-SideNav--itemText-27"
                >
                  Dashboard
                </div>
                <span
                  className="MuiTouchRipple-root-123"
                />
              </div>
              <div
                className="MuiButtonBase-root-115 MuiListItem-root-134 MuiListItem-default-137 MuiListItem-gutters-141 MuiListItem-button-142 withRouter-SideNav--itemLink-24 withRouter-SideNav--itemLinkClose-25"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-117 MuiListItemIcon-root-144 withRouter-SideNav--itemIcon-26"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <g>
                    <path
                      d="M19 2h-4.18C14.4.84 13.3 0 12 0c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm7 18H5V4h2v3h10V4h2v16z"
                    />
                  </g>
                </svg>
                <div
                  className="MuiListItemText-root-145 withRouter-SideNav--itemText-27"
                >
                  Request Tracker
                </div>
                <span
                  className="MuiTouchRipple-root-123"
                />
              </div>
            </ul>
            <div
              className="withRouter-SideNav--linkDiv-12"
            >
              <hr
                className="MuiDivider-root-151"
              />
              <button
                aria-label="Settings"
                className="MuiButtonBase-root-115 MuiIconButton-root-109 MuiIconButton-colorInherit-110"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="0"
                type="button"
              >
                <span
                  className="MuiIconButton-label-114"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-117"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <g>
                      <path
                        d="M19.43 12.98c.04-.32.07-.64.07-.98s-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46c-.12-.22-.39-.3-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65C14.46 2.18 14.25 2 14 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1c-.23-.09-.49 0-.61.22l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98s.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46c.12.22.39.3.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.23.09.49 0 .61-.22l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zM12 15.5c-1.93 0-3.5-1.57-3.5-3.5s1.57-3.5 3.5-3.5 3.5 1.57 3.5 3.5-1.57 3.5-3.5 3.5z"
                      />
                    </g>
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-123"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <header
    className="MuiPaper-root-45 MuiPaper-elevation4-51 MuiAppBar-root-37 MuiAppBar-positionAbsolute-39 MuiAppBar-colorPrimary-43 Shell-appBar-2"
  >
    <div
      className="MuiToolbar-root-72 Shell-toolbar-9"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root-115 MuiIconButton-root-109 MuiIconButton-colorInherit-110 Shell-menuButton-4"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-114"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-117"
            color={undefined}
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-123"
        />
      </button>
      <h2
        className="MuiTypography-root-74 MuiTypography-title-80 MuiTypography-colorInherit-93 MuiTypography-noWrap-90"
      >
        Redirect
      </h2>
    </div>
  </header>
  <div
    className="Shell-content-10"
  >
    <div
      className="Shell-toolbar-9"
    />
    <div
      className="Shell-actualContent-11"
    />
  </div>
</div>
`;
